src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1852,13 +1837,85 @@
     </Field>
     <Field Name="Value" Type="ExpressionSyntax"/>
   </Node>
+
+  <AbstractNode Name="VariableComponentSyntax" Base="CSharpSyntaxNode">
+  </AbstractNode>
+  <Node Name="TypedVariableComponentSyntax" Base="VariableComponentSyntax">
+    <Kind Name="TypedVariableComponent"/>
+    <Field Name="Type" Type="TypeSyntax"/>
+    <Field Name="Designation" Type="VariableDesignationSyntax"/>
+  </Node>
+  <Node Name="TupleDeconstructionVariableComponentSyntax" Base="VariableComponentSyntax">
+    <Kind Name="TupleDeconstructionVariableComponent"/>
+    <Field Name="OpenParenToken" Type="SyntaxToken">
+      <Kind Name="OpenParenToken"/>
+    </Field>
+    <Field Name="Variables" Type="SeparatedSyntaxList&lt;VariableComponentSyntax&gt;"/>
+    <Field Name="CloseParenToken" Type="SyntaxToken">
+      <Kind Name="CloseParenToken"/>
+    </Field>
+  </Node>
+  <!--
+  <Node Name="WildcardVariableComponentSyntax" Base="VariableComponentSyntax">
+    <Kind Name="WildcardVariableComponent"/>
+    <Field Name="AsteriskToken" Type="SyntaxToken">
+      <Kind Name="AsteriskToken"/>
+    </Field>
+  </Node>
+  -->
+
+  <AbstractNode Name="VariableDesignationSyntax" Base="CSharpSyntaxNode">
+  </AbstractNode>
+  <Node Name="SingleVariableDesignationSyntax" Base="VariableDesignationSyntax">
```

Comment:
An initializer is forbidden as part of the declaration in all the contexts VariableDesignationSyntax is used to model the grammar, so that doesn't represent a factoring of the language grammar. The new node DeconstructionDeclarationAssignmentSyntax represents a VariableDesignationSyntax along with the initializer, and is capable of representing "type v = expr".

---
In reply to: [73446342](https://github.com/dotnet/roslyn/pull/12917#discussion_r73446342) [](ancestors = 73446342)